app/uk/gov/hmrc/gform/obligation/ObligationService.scala

```diff
@@ -28,47 +26,83 @@ import uk.gov.hmrc.gform.sharedmodel._
 import uk.gov.hmrc.gform.sharedmodel.form.{ Form, FormId, UserData }
 import uk.gov.hmrc.gform.sharedmodel.formtemplate._
 import uk.gov.hmrc.http.HeaderCarrier
+import cats.implicits._
 
 import scala.concurrent.{ ExecutionContext, Future }
 
 class ObligationService(gformConnector: GformConnector) {
 
   val stringToDate = new SimpleDateFormat("yyyy-MM-dd")
 
-  private def lookupObligationsMultiple(
+  private def makeAllInfoList(
+    id: HmrcTaxPeriod,
+    evaluatedId: HmrcTaxPeriodWithEvaluatedId,
+    obligation: List[ObligationDetail]) =
+    obligation.map(
+      i =>
+        TaxPeriodInformation(
+          id,
+          evaluatedId.idNumberValue,
+          i.inboundCorrespondenceFromDate,
+          i.inboundCorrespondenceToDate,
+          i.periodKey))
+
+  private def updatedObligations(
+    idNumbers: NonEmptyList[HmrcTaxPeriodWithEvaluatedId],
+    taxResponses: List[TaxResponse]): List[TaxPeriodInformation] =
+    for {
+      taxResponse  <- taxResponses
+      obligDetails <- taxResponse.obligation.obligations
+      idNumber     <- idNumbers.find(hTPWithId => hTPWithId.hmrcTaxPeriod == taxResponse.id).toList
```

Comment:
We can get `idNumber` directly from `taxResponse` if we changed `TaxResponse` from:

```
case class TaxResponse(hmrcTaxPeriod: HmrcTaxPeriod, obligation: Obligation)
```
to
```
case class TaxResponse(id: HmrcTaxPeriodWithEvaluatedId, obligation: Obligation)
```